commands/miscellaneous/getStudentProfile.js

```diff
@@ -6,61 +6,60 @@ const ALL_GUILD_IDS = [BATId, BALId, BAGId, BADId, guildId];
 
 module.exports = {
   data: new SlashCommandBuilder()
-    .setName('getstudentprofile')
-    .setDescription('Gets info of a students across BA servers')
-    .setDefaultPermission(false)
-    .addUserOption(option =>
-      option.setName('user')
-        .setDescription('The user you want to get their profile.')
-        .setRequired(true)),
+      .setName('getstudentprofile')
+      .setDescription('Gets info of a students across BA servers')
+      .setDefaultPermission(false)
+      .addUserOption(option =>
+          option.setName('user')
+              .setDescription('The user you want to get their profile.')
+              .setRequired(true)),
   async execute(interaction) {
     const options = interaction.options;
     const user = options.getUser('user');
     const fields = [];
 
-    const formatStudentInfo = (guildName, value) => {
+    const formatField = (displayName, value) => {
       const info = {
-        name: `Roles at ${guildName}`,
+        name: `${displayName}: `,
         value: `${value}`,
-        inline: true
+        inline: false
       };
       return info;
     };
 
     await interaction.deferReply();
+    let userInGuild = null;
     ALL_GUILD_IDS.forEach(guildId => {
       const guild = interaction.client.guilds.cache.get(guildId);
       if (guildId && !guild) return;
-      const userInGuild = guild.members.cache.find(u => u.id === user.id);
+      userInGuild = guild.members.cache.find(u => u.id === user.id);
       if (!userInGuild) return;
       let roles = userInGuild.roles.cache.filter(role => role.name != '@everyone');
       if (interaction.guild == guildId) {
         roles = roles.map(role => `<@&${role.id}>`).join('\n');
       } else {
         roles = roles.map(role => role.name).join('\n');
       }
-      const joinedDate = new Date(userInGuild.joinedAt).toLocaleString('ko-KR', { timeZone: 'UTC' });
-      const result = `${roles}\n\n **Joined at:** ${joinedDate} UTC`;
-      if (result.length > 1024) {
-        const results = splitText(result, '\n');
-        results.forEach(result => {
-          fields.push(formatStudentInfo(guild.name, result));
+      if (roles.length > 1024) {
+        roles.forEach(role => {
+          fields.push(formatField(`Roles in ${guild.name}`, role));
         });
       } else {
-        fields.push(formatStudentInfo(guild.name, result));
+        fields.push(formatField(`Roles in ${guild.name}`, roles));
       }
     });
+    const displayName = userInGuild.displayName;
```

Comment:
this is just gonna be the displayName for the user in whatever the last guild in the loop is. probably fine since they should all match. but we could also find the user in the main guild by using `guildId` from config and then set the displayName to that displayName. or show a nickname per server, which might be nice cause then you'd easily catch if one wasn't matching?